From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VSCodium <scripts@vscodium.com>
Date: Sat, 1 Jan 2000 00:00:00 +0000
Subject: 00-ui-update-emulated-urls


diff --git a/src/vs/workbench/electron-browser/window.ts b/src/vs/workbench/electron-browser/window.ts
index b60c0adc62c0fea363ebc9f38436524724e15cff..10935eaf837fef44632e640480f04211f46d58aa 100644
--- a/src/vs/workbench/electron-browser/window.ts
+++ b/src/vs/workbench/electron-browser/window.ts
@@ -261,8 +261,8 @@ export class NativeWindow extends BaseWindow {
 					label: localize('downloadArmBuild', "Download"),
 					run: () => {
 						const quality = this.productService.quality;
-						const stableURL = 'https://code.visualstudio.com/docs/?dv=osx';
-						const insidersURL = 'https://code.visualstudio.com/docs/?dv=osx&build=insiders';
+						const stableURL = 'https://github.com/!!GH_REPO_PATH!!/releases/latest';
+						const insidersURL = 'https://github.com/!!GH_REPO_PATH!!-insiders/releases/latest';
 						this.openerService.open(quality === 'stable' ? stableURL : insidersURL);
 					}
 				}],
