diff --git a/build/darwin/sign.js b/build/darwin/sign.js
index dff30fd..df48bee 100644
--- a/build/darwin/sign.js
+++ b/build/darwin/sign.js
@@ -56,5 +56,7 @@ async function main(buildDir) {
         ignore: (filePath) => {
+            const ext = path_1.default.extname(filePath);
             return filePath.includes(gpuHelperAppName) ||
                 filePath.includes(rendererHelperAppName) ||
-                filePath.includes(pluginHelperAppName);
+                filePath.includes(pluginHelperAppName) ||
+                ext == '.asar' || ext == '.dat' || ext == '.gif' || ext == '.icns' || ext == '.ico' || ext == '.json' || ext == '.mp3' || ext == '.nib' || ext == '.pak' || ext == '.png' || ext == '.scpt' || ext == '.ttf' || ext == '.wasm' || ext == '.woff' || ext == '.woff2';
         }
@@ -83,3 +85,3 @@ async function main(buildDir) {
             '-string',
-            'An application in Visual Studio Code wants to use AppleScript.',
+            'An application in Codex wants to use AppleScript.',
             `${infoPlistPath}`
@@ -90,3 +92,3 @@ async function main(buildDir) {
             '-string',
-            'An application in Visual Studio Code wants to use the Microphone.',
+            'An application in Codex wants to use the Microphone.',
             `${infoPlistPath}`
@@ -97,3 +99,3 @@ async function main(buildDir) {
             '-string',
-            'An application in Visual Studio Code wants to use the Camera.',
+            'An application in Codex wants to use the Camera.',
             `${infoPlistPath}`
