diff --git a/build/gulpfile.reh.js b/build/gulpfile.reh.js
index 2a0c236..0f10f17 100644
--- a/build/gulpfile.reh.js
+++ b/build/gulpfile.reh.js
@@ -256,3 +256,3 @@ function packageTask(type, platform, arch, sourceFolderName, destinationFolderNa
 		const productJsonStream = gulp.src(['product.json'], { base: '.' })
-			.pipe(json({ commit, date, version, release }));
+			.pipe(json({ commit, date, version, release, serverDownloadUrlTemplate: 'https://github.com/BiblioNexus-Foundation/codex-insiders/releases/download/${version}.${release}-insider/codex-reh-${os}-${arch}-${version}.${release}-insider.tar.gz' }));
 
diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 0ca2cfd..9bf83a7 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -244,3 +244,3 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 		const date = new Date().toISOString();
-		const productJsonUpdate = { commit, date, checksums, version, release };
+		const productJsonUpdate = { commit, date, checksums, version, release, serverDownloadUrlTemplate: 'https://github.com/BiblioNexus-Foundation/codex-insiders/releases/download/${version}.${release}-insider/codex-reh-${os}-${arch}-${version}.${release}-insider.tar.gz' };
 
