diff --git a/src/vs/workbench/contrib/remote/browser/remote.contribution.ts b/src/vs/workbench/contrib/remote/browser/remote.contribution.ts
index 932a92c..afa3272 100644
--- a/src/vs/workbench/contrib/remote/browser/remote.contribution.ts
+++ b/src/vs/workbench/contrib/remote/browser/remote.contribution.ts
@@ -14,2 +14,4 @@ import { AutomaticPortForwarding, ForwardedPortsView, PortRestore } from 'vs/wor
 import { RemoteStartEntry } from 'vs/workbench/contrib/remote/browser/remoteStartEntry';
+import { CommandsRegistry } from 'vs/platform/commands/common/commands';
+import { localize } from 'vs/nls';
 
@@ -25 +27,13 @@ workbenchContributionsRegistry.registerWorkbenchContribution(RemoteMarkers, Life
 workbenchContributionsRegistry.registerWorkbenchContribution(RemoteStartEntry, LifecyclePhase.Starting);
+
+CommandsRegistry.registerCommand({
+	id: 'remote.serverDownloadUrlTemplate',
+	description: {
+		description: localize('remote.serverDownloadUrlTemplate', 'Provides the url template to download the server archive'),
+		args: [],
+		returns: 'string',
+	},
+	handler: async (_) => {
+		return 'https://github.com/VSCodium/vscodium-insiders/releases/download/${version}.${release}-insider/vscodium-reh-${os}-${arch}-${version}.${release}-insider.tar.gz';
+	}
+});
\ No newline at end of file
