diff --git a/src/vs/platform/windows/electron-main/windowImpl.ts b/src/vs/platform/windows/electron-main/windowImpl.ts
index 3b8a644..71ea0ad 100644
--- a/src/vs/platform/windows/electron-main/windowImpl.ts
+++ b/src/vs/platform/windows/electron-main/windowImpl.ts
@@ -213,4 +213,2 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 				useSandbox = true;
-			} else {
-				useSandbox = typeof this.productService.quality === 'string' && this.productService.quality !== 'stable';
 			}
