diff --git a/build/gulpfile.vscode.win32.js b/build/gulpfile.vscode.win32.js
index 4a9a644..45c1e2b 100644
--- a/build/gulpfile.vscode.win32.js
+++ b/build/gulpfile.vscode.win32.js
@@ -114,7 +114,7 @@ function buildWin32Setup(arch, target) {
 
-		if (quality === 'insider') {
-			definitions['AppxPackage'] = `code_insiders_explorer_${arch}.appx`;
-			definitions['AppxPackageFullname'] = `Microsoft.${product.win32RegValueName}_1.0.0.0_neutral__8wekyb3d8bbwe`;
-			definitions['AppxPackageName'] = `Microsoft.${product.win32RegValueName}`;
-		}
+		// if (quality === 'insider') {
+		// 	definitions['AppxPackage'] = `code_insiders_explorer_${arch}.appx`;
+		// 	definitions['AppxPackageFullname'] = `Microsoft.${product.win32RegValueName}_1.0.0.0_neutral__8wekyb3d8bbwe`;
+		// 	definitions['AppxPackageName'] = `Microsoft.${product.win32RegValueName}`;
+		// }
 
