diff --git a/src/main.js b/src/main.js
index 63f8c56..a6b7d44 100644
--- a/src/main.js
+++ b/src/main.js
@@ -575,6 +575,10 @@ async function resolveNlsConfiguration() {
 		 */
-		// @ts-ignore API not yet available in the official Electron
-		let appLocale = ((product.quality === 'insider' || product.quality === 'exploration') && app?.getPreferredSystemLanguages()?.length) ?
+		let appLocale = app.getLocale();
+		if (product.quality === 'insider' || product.quality === 'exploration') {
 			// @ts-ignore API not yet available in the official Electron
-			app.getPreferredSystemLanguages()[0] : app.getLocale();
+			if (app.getPreferredSystemLanguages && app?.getPreferredSystemLanguages()?.length) {
+				// @ts-ignore API not yet available in the official Electron
+				appLocale = app.getPreferredSystemLanguages()[0];
+			}
+		}
 		if (!appLocale) {
