From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VSCodium <scripts@vscodium.com>
Date: Sat, 1 Jan 2000 00:00:00 +0000
Subject: 00-build-min-version-win7


diff --git a/build/win32/code.iss b/build/win32/code.iss
index b78bc0c5d4134f5e7748177d81c5f9754482162b..35f0ccaa1b8e59bba32e4c44a02a93622ecc8522 100644
--- a/build/win32/code.iss
+++ b/build/win32/code.iss
@@ -26,7 +26,7 @@ SetupIconFile={#RepoDir}\resources\win32\code.ico
 UninstallDisplayIcon={app}\{#ExeBasename}.exe
 ChangesEnvironment=true
 ChangesAssociations=true
-MinVersion=10.0
+MinVersion=6.1.7600
 SourceDir={#SourceDir}
 AppVersion={#Version}
 VersionInfoVersion={#RawVersion}
