From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VSCodium <scripts@vscodium.com>
Date: Sat, 1 Jan 2000 00:00:00 +0000
Subject: 00-build-min-version-win7


diff --git a/build/win32/code.iss b/build/win32/code.iss
index 5b4306143f6c7a1268c7a646c271e79a1ab374cf..2da54e85a33c90ce6789aab52e139141777a7b8c 100644
--- a/build/win32/code.iss
+++ b/build/win32/code.iss
@@ -26,7 +26,7 @@ SetupIconFile={#RepoDir}\resources\win32\code.ico
 UninstallDisplayIcon={app}\{#ExeBasename}.exe
 ChangesEnvironment=true
 ChangesAssociations=true
-MinVersion=10.0
+MinVersion=6.1.7600
 SourceDir={#SourceDir}
 AppVersion={#Version}
 VersionInfoVersion={#RawVersion}
