From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VSCodium <scripts@vscodium.com>
Date: Sat, 1 Jan 2000 00:00:00 +0000
Subject: 00-remote-use-open-ext


diff --git a/resources/win32/bin/code.sh b/resources/win32/bin/code.sh
index bcf31892c2345e12a4f4f3a08dc5ba268e5f66b0..4f0653998a2856521a7508ba326ba76d714f3bff 100644
--- a/resources/win32/bin/code.sh
+++ b/resources/win32/bin/code.sh
@@ -11,6 +11,7 @@ APP_NAME="@@APPNAME@@"
 QUALITY="@@QUALITY@@"
 NAME="@@NAME@@"
 SERVERDATAFOLDER="@@SERVERDATAFOLDER@@"
+VERSION="@@VERSION@@"
 VSCODE_PATH="$(dirname "$(dirname "$(realpath "$0")")")"
 ELECTRON="$VSCODE_PATH/$NAME.exe"
 
@@ -41,7 +42,7 @@ if [ $IN_WSL = true ]; then
 	CLI=$(wslpath -m "$VSCODE_PATH/resources/app/out/cli.js")
 
 	# use the Remote WSL extension if installed
-	WSL_EXT_ID="ms-vscode-remote.remote-wsl"
+	WSL_EXT_ID="${VSCODE_WSL_EXTENSION_ID:-jeanp413.open-remote-wsl}"
 
 	ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" --locate-extension $WSL_EXT_ID >/tmp/remote-wsl-loc.txt 2>/dev/null </dev/null
 	WSL_EXT_WLOC=$(cat /tmp/remote-wsl-loc.txt)
@@ -49,7 +50,7 @@ if [ $IN_WSL = true ]; then
 	if [ -n "$WSL_EXT_WLOC" ]; then
 		# replace \r\n with \n in WSL_EXT_WLOC
 		WSL_CODE=$(wslpath -u "${WSL_EXT_WLOC%%[[:cntrl:]]}")/scripts/wslCode.sh
-		"$WSL_CODE" "$COMMIT" "$QUALITY" "$ELECTRON" "$APP_NAME" "$SERVERDATAFOLDER" "$@"
+		"$WSL_CODE" "$COMMIT" "$QUALITY" "$ELECTRON" "$APP_NAME" "$SERVERDATAFOLDER" "$VERSION" "$@"
 		exit $?
 	fi
 
diff --git a/resources/win32/versioned/bin/code.sh b/resources/win32/versioned/bin/code.sh
index 2443d965ca79fef361e1d58318e229bfbf304291..b11fd5b4204f1571a20d05a4498f8788f96e30a2 100644
--- a/resources/win32/versioned/bin/code.sh
+++ b/resources/win32/versioned/bin/code.sh
@@ -11,6 +11,7 @@ APP_NAME="@@APPNAME@@"
 QUALITY="@@QUALITY@@"
 NAME="@@NAME@@"
 SERVERDATAFOLDER="@@SERVERDATAFOLDER@@"
+VERSION="@@VERSION@@"
 VERSIONFOLDER="@@VERSIONFOLDER@@"
 VSCODE_PATH="$(dirname "$(dirname "$(realpath "$0")")")"
 ELECTRON="$VSCODE_PATH/$NAME.exe"
@@ -42,7 +43,7 @@ if [ $IN_WSL = true ]; then
 	CLI=$(wslpath -m "$VSCODE_PATH/$VERSIONFOLDER/resources/app/out/cli.js")
 
 	# use the Remote WSL extension if installed
-	WSL_EXT_ID="ms-vscode-remote.remote-wsl"
+	WSL_EXT_ID="${VSCODE_WSL_EXTENSION_ID:-jeanp413.open-remote-wsl}"
 
 	ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" --locate-extension $WSL_EXT_ID >/tmp/remote-wsl-loc.txt 2>/dev/null </dev/null
 	WSL_EXT_WLOC=$(cat /tmp/remote-wsl-loc.txt)
@@ -50,7 +51,7 @@ if [ $IN_WSL = true ]; then
 	if [ -n "$WSL_EXT_WLOC" ]; then
 		# replace \r\n with \n in WSL_EXT_WLOC
 		WSL_CODE=$(wslpath -u "${WSL_EXT_WLOC%%[[:cntrl:]]}")/scripts/wslCode.sh
-		"$WSL_CODE" "$COMMIT" "$QUALITY" "$ELECTRON" "$APP_NAME" "$SERVERDATAFOLDER" "$@"
+		"$WSL_CODE" "$COMMIT" "$QUALITY" "$ELECTRON" "$APP_NAME" "$SERVERDATAFOLDER" "$VERSION" "$@"
 		exit $?
 	fi
 
diff --git a/scripts/code.sh b/scripts/code.sh
index 16fdefde55208d12447773a32d329b17b6a6f112..00589cdeff374e2dd529f0a728e01333443c3cf7 100755
--- a/scripts/code.sh
+++ b/scripts/code.sh
@@ -63,7 +63,7 @@ function code-wsl()
 		local CWD=$(pwd)
 		cd $ROOT
 		export WSLENV=ELECTRON_RUN_AS_NODE/w:VSCODE_DEV/w:$WSLENV
-		local WSL_EXT_ID="ms-vscode-remote.remote-wsl"
+		local WSL_EXT_ID="${VSCODE_WSL_EXTENSION_ID:-jeanp413.open-remote-wsl}"
 		local WSL_EXT_WLOC=$(echo "" | VSCODE_DEV=1 ELECTRON_RUN_AS_NODE=1 "$ROOT/.build/electron/Code - OSS.exe" "out/cli.js" --locate-extension $WSL_EXT_ID)
 		cd $CWD
 		if [ -n "$WSL_EXT_WLOC" ]; then
