From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VSCodium <scripts@vscodium.com>
Date: Sat, 1 Jan 2000 00:00:00 +0000
Subject: 71-build-disable-appx


diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
index 79aa99bf81f67ceb9cddf2c3b35018e7055771e3..aca2a56ca9975d4839bd594d966fabff600a0f02 100644
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -541,22 +541,22 @@ function packageTask(platform: string, arch: string, sourceFolderName: string, d
 			result = es.merge(result, gulp.src('.build/policies/win32/**', { base: '.build/policies/win32' })
 				.pipe(rename(f => f.dirname = `policies/${f.dirname}`)));
 
-			if (quality === 'stable' || quality === 'insider') {
-				result = es.merge(result, gulp.src('.build/win32/appx/**', { base: '.build/win32' }));
-				const rawVersion = version.replace(/-\w+$/, '').split('.');
-				const appxVersion = `${rawVersion[0]}.0.${rawVersion[1]}.${rawVersion[2].slice(1)}`;
-				result = es.merge(result, gulp.src('resources/win32/appx/AppxManifest.xml', { base: '.' })
-					.pipe(replace('@@AppxPackageName@@', product.win32AppUserModelId))
-					.pipe(replace('@@AppxPackageVersion@@', appxVersion))
-					.pipe(replace('@@AppxPackageDisplayName@@', product.nameLong))
-					.pipe(replace('@@AppxPackageDescription@@', product.win32NameVersion))
-					.pipe(replace('@@ApplicationIdShort@@', product.win32RegValueName))
-					.pipe(replace('@@ApplicationExe@@', product.nameShort + '.exe'))
-					.pipe(replace('@@FileExplorerContextMenuID@@', quality === 'stable' ? 'OpenWithCode' : 'OpenWithCodeInsiders'))
-					.pipe(replace('@@FileExplorerContextMenuCLSID@@', (product as { win32ContextMenu?: Record<string, { clsid: string }> }).win32ContextMenu![arch].clsid))
-					.pipe(replace('@@FileExplorerContextMenuDLL@@', `${quality === 'stable' ? 'code' : 'code_insider'}_explorer_command_${arch}.dll`))
-					.pipe(rename(f => f.dirname = `appx/manifest`)));
-			}
+			// if (quality === 'stable' || quality === 'insider') {
+			// 	result = es.merge(result, gulp.src('.build/win32/appx/**', { base: '.build/win32' }));
+			// 	const rawVersion = version.replace(/-\w+$/, '').split('.');
+			// 	const appxVersion = `${rawVersion[0]}.0.${rawVersion[1]}.${rawVersion[2].slice(1)}`;
+			// 	result = es.merge(result, gulp.src('resources/win32/appx/AppxManifest.xml', { base: '.' })
+			// 		.pipe(replace('@@AppxPackageName@@', product.win32AppUserModelId))
+			// 		.pipe(replace('@@AppxPackageVersion@@', appxVersion))
+			// 		.pipe(replace('@@AppxPackageDisplayName@@', product.nameLong))
+			// 		.pipe(replace('@@AppxPackageDescription@@', product.win32NameVersion))
+			// 		.pipe(replace('@@ApplicationIdShort@@', product.win32RegValueName))
+			// 		.pipe(replace('@@ApplicationExe@@', product.nameShort + '.exe'))
+			// 		.pipe(replace('@@FileExplorerContextMenuID@@', quality === 'stable' ? 'OpenWithCode' : 'OpenWithCodeInsiders'))
+			// 		.pipe(replace('@@FileExplorerContextMenuCLSID@@', (product as { win32ContextMenu?: Record<string, { clsid: string }> }).win32ContextMenu![arch].clsid))
+			// 		.pipe(replace('@@FileExplorerContextMenuDLL@@', `${quality === 'stable' ? 'code' : 'code_insider'}_explorer_command_${arch}.dll`))
+			// 		.pipe(rename(f => f.dirname = `appx/manifest`)));
+			// }
 		} else if (platform === 'linux') {
 			result = es.merge(result, gulp.src('resources/linux/bin/code.sh', { base: '.' })
 				.pipe(replace('@@PRODNAME@@', product.nameLong))
diff --git a/build/gulpfile.vscode.win32.ts b/build/gulpfile.vscode.win32.ts
index c43819dfbfa279d1502d19edfccd0cac5c4cf9ac..a3037188e0c7b59b0a579bb92fd56683ddd3cea1 100644
--- a/build/gulpfile.vscode.win32.ts
+++ b/build/gulpfile.vscode.win32.ts
@@ -113,15 +113,15 @@ function buildWin32Setup(arch: string, target: string): task.CallbackTask {
 			Quality: quality
 		};
 
-		if (quality === 'stable' || quality === 'insider') {
-			definitions['AppxPackage'] = `${product.applicationName.replaceAll('-', '_')}_${arch}.appx`;
-			definitions['AppxPackageDll'] = `${product.applicationName.replaceAll('-', '_')}_explorer_command_${arch}.dll`;
-			definitions['AppxPackageName'] = `${product.win32AppUserModelId}`;
-			const ctxMenu = (product as { win32ContextMenu?: Record<string, { clsid: string }> }).win32ContextMenu;
-			if (ctxMenu && ctxMenu[arch]) {
-				definitions['FileExplorerContextMenuCLSID'] = ctxMenu[arch].clsid;
-			}
-		}
+		// if (quality === 'stable' || quality === 'insider') {
+		// 	definitions['AppxPackage'] = `${product.applicationName.replaceAll('-', '_')}_${arch}.appx`;
+		// 	definitions['AppxPackageDll'] = `${product.applicationName.replaceAll('-', '_')}_explorer_command_${arch}.dll`;
+		// 	definitions['AppxPackageName'] = `${product.win32AppUserModelId}`;
+		// 	const ctxMenu = (product as { win32ContextMenu?: Record<string, { clsid: string }> }).win32ContextMenu;
+		// 	if (ctxMenu && ctxMenu[arch]) {
+		// 		definitions['FileExplorerContextMenuCLSID'] = ctxMenu[arch].clsid;
+		// 	}
+		// }
 
 		fs.writeFileSync(productJsonPath, JSON.stringify(productJson, undefined, '\t'));
 
