diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
index d3ab651..6e2d01a 100644
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -477,2 +477,4 @@ function patchWin32DependenciesTask(destinationFolderName: string) {
 
+		console.log(cwd);
+
 		await Promise.all(deps.map(async dep => {
@@ -480,2 +482,4 @@ function patchWin32DependenciesTask(destinationFolderName: string) {
 
+			console.log(path.join(cwd, dep));
+
 			await rcedit(path.join(cwd, dep), {
diff --git a/extensions/microsoft-authentication/extension.webpack.config.js b/extensions/microsoft-authentication/extension.webpack.config.js
index a46d5a5..14f2dea 100644
--- a/extensions/microsoft-authentication/extension.webpack.config.js
+++ b/extensions/microsoft-authentication/extension.webpack.config.js
@@ -40,2 +40,3 @@ if (
 ) {
+	console.log('Copy native files');
 	plugins.push(new CopyWebpackPlugin({
@@ -49,2 +50,4 @@ if (
 	}));
+} else {
+	console.log('Do not copy native files');
 }
