diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
index d3ab651..105dc32 100644
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -218,3 +218,7 @@ function packageTask(platform: string, arch: string, sourceFolderName: string, d
 
-	const task = () => {
+	const task = async () => {
+		console.log(await glob('**/*.node', { cwd: destination }))
+		console.log(await glob('**/*.node', { cwd: path.join('.', sourceFolderName) }))
+		console.log(await glob('**/*.node', { cwd: path.join(process.cwd(), 'node_modules') }))
+
 		const out = sourceFolderName;
@@ -462,3 +466,3 @@ function packageTask(platform: string, arch: string, sourceFolderName: string, d
 
-		return result.pipe(vfs.dest(destination));
+		return Promise.resolve(result.pipe(vfs.dest(destination)));
 	};
@@ -477,2 +481,4 @@ function patchWin32DependenciesTask(destinationFolderName: string) {
 
+		console.log(cwd);
+
 		await Promise.all(deps.map(async dep => {
@@ -480,2 +486,4 @@ function patchWin32DependenciesTask(destinationFolderName: string) {
 
+			console.log(path.join(cwd, dep));
+
 			await rcedit(path.join(cwd, dep), {
diff --git a/extensions/microsoft-authentication/extension.webpack.config.js b/extensions/microsoft-authentication/extension.webpack.config.js
index a46d5a5..14f2dea 100644
--- a/extensions/microsoft-authentication/extension.webpack.config.js
+++ b/extensions/microsoft-authentication/extension.webpack.config.js
@@ -40,2 +40,3 @@ if (
 ) {
+	console.log('Copy native files');
 	plugins.push(new CopyWebpackPlugin({
@@ -49,2 +50,4 @@ if (
 	}));
+} else {
+	console.log('Do not copy native files');
 }
