diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 25d8916..11336b7 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -417,5 +417,5 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
 					.pipe(replace('@@ApplicationExe@@', product.nameShort + '.exe'))
-					.pipe(replace('@@FileExplorerContextMenuID@@', quality === 'stable' ? 'OpenWithCode' : 'OpenWithCodeInsiders'))
+					.pipe(replace('@@FileExplorerContextMenuID@@', `OpenWith${product.win32RegValueName}`))
 					.pipe(replace('@@FileExplorerContextMenuCLSID@@', product.win32ContextMenu[arch].clsid))
-					.pipe(replace('@@FileExplorerContextMenuDLL@@', `${quality === 'stable' ? 'code' : 'code_insider'}_explorer_command_${arch}.dll`))
+					.pipe(replace('@@FileExplorerContextMenuDLL@@', `${product.applicationName.replace(/-/g, '_')}_explorer_command_${arch}.dll`))
 					.pipe(rename(f => f.dirname = `appx/manifest`)));
