diff --git a/build/npm/dirs.ts b/build/npm/dirs.ts
index 48d76e2..270b078 100644
--- a/build/npm/dirs.ts
+++ b/build/npm/dirs.ts
@@ -53,4 +53,2 @@ export const dirs = [
 	'extensions/vscode-test-resolver',
-	'remote',
-	'remote/web',
 	'test/automation',
@@ -66,4 +64,2 @@ if (existsSync(`${import.meta.dirname}/../../.build/distro/npm`)) {
 	dirs.push('.build/distro/npm');
-	dirs.push('.build/distro/npm/remote');
-	dirs.push('.build/distro/npm/remote/web');
 }
