diff --git a/build/gulpfile.extensions.ts b/build/gulpfile.extensions.ts
index fdf68387..f8adb5e1 100644
--- a/build/gulpfile.extensions.ts
+++ b/build/gulpfile.extensions.ts
@@ -24,2 +24,3 @@ import * as util from './lib/util.ts';
 import watcher from './lib/watch/index.ts';
+import product from '../product.json' with { type: 'json' };
 
@@ -99,3 +100,3 @@ const compilations = [
 
-const getBaseUrl = (out: string) => `https://main.vscode-cdn.net/sourcemaps/${commit}/${out}`;
+const getBaseUrl = (out: string) => `https://github.com/VSCodium/sourcemaps/releases/download/${product.quality}-${commit}/extensions-${out.replaceAll('/', '-')}`;
 
diff --git a/build/gulpfile.reh.ts b/build/gulpfile.reh.ts
index c643cf99..31bb52ee 100644
--- a/build/gulpfile.reh.ts
+++ b/build/gulpfile.reh.ts
@@ -536,3 +536,3 @@ function prepareCopilotRipgrepShimTaskREH(platform: string, arch: string, destin
 	const esbuildBundleTask = task.define(`esbuild-vscode-${type}`, () => runEsbuildBundle(`out-vscode-${type}`, false, true, target));
-	const esbuildBundleMinTask = task.define(`esbuild-vscode-${type}-min`, () => runEsbuildBundle(`out-vscode-${type}-min`, true, true, target, `https://main.vscode-cdn.net/sourcemaps/${commit}/core`));
+	const esbuildBundleMinTask = task.define(`esbuild-vscode-${type}-min`, () => runEsbuildBundle(`out-vscode-${type}-min`, true, true, target, `https://github.com/VSCodium/sourcemaps/releases/download/${product.quality}-${commit}/core`));
 
diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
index 95ae1790..46af72f8 100644
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -61,3 +61,3 @@ function getLockedPackageVersion(packageName: string): string {
 
-const sourceMappingURLBase = `https://main.vscode-cdn.net/sourcemaps/${commit}`;
+const sourceMappingURLBase = `https://github.com/VSCodium/sourcemaps/releases/download/${product.quality}-${commit}`;
 const isCI = !!process.env['CI'] || !!process.env['BUILD_ARTIFACTSTAGINGDIRECTORY'] || !!process.env['GITHUB_WORKSPACE'];
diff --git a/build/gulpfile.vscode.web.ts b/build/gulpfile.vscode.web.ts
index 6afedb64..9b68bf0a 100644
--- a/build/gulpfile.vscode.web.ts
+++ b/build/gulpfile.vscode.web.ts
@@ -28,3 +28,3 @@ const version = (quality && quality !== 'stable') ? `${packageJson.version}-${qu
 
-const sourceMappingURLBase = `https://main.vscode-cdn.net/sourcemaps/${commit}`;
+const sourceMappingURLBase = `https://github.com/VSCodium/sourcemaps/releases/download/${product.quality}-${commit}`;
 const esbuildBundleVSCodeWebTask = task.define('esbuild-vscode-web', () => runEsbuildBundle('out-vscode-web', false, true, 'web'));
diff --git a/build/lib/extensions.ts b/build/lib/extensions.ts
index 10bd9fbd..d6494c55 100644
--- a/build/lib/extensions.ts
+++ b/build/lib/extensions.ts
@@ -29,4 +29,2 @@ const require = createRequire(import.meta.url);
 const root = path.dirname(path.dirname(import.meta.dirname));
-// const commit = getVersion(root);
-// const sourceMappingURLBase = `https://main.vscode-cdn.net/sourcemaps/${commit}`;
 
diff --git a/build/lib/util.ts b/build/lib/util.ts
index 82135b31..58385d41 100644
--- a/build/lib/util.ts
+++ b/build/lib/util.ts
@@ -286,4 +286,4 @@ export function rewriteSourceMappingURL(sourceMappingURLBase: string): NodeJS.Re
 			const contents = (f.contents as Buffer).toString('utf8');
-			const str = `//# sourceMappingURL=${sourceMappingURLBase}/${path.dirname(f.relative).replace(/\\/g, '/')}/$1`;
-			f.contents = Buffer.from(contents.replace(/\n\/\/# sourceMappingURL=(.*)$/gm, str));
+			const fp = path.dirname(f.relative).replace(/\\/g, '/').replaceAll('/', '-');
+			f.contents = Buffer.from(contents.replace(/\n\/\/# sourceMappingURL=(.*)$/gm, (_m, p) => `//# sourceMappingURL=${sourceMappingURLBase}/${fp}-${p.replaceAll('/', '-')}`));
 			return f;
