diff --git a/build/lib/builtInExtensions.ts b/build/lib/builtInExtensions.ts
index de687da4..2436a1fc 100644
--- a/build/lib/builtInExtensions.ts
+++ b/build/lib/builtInExtensions.ts
@@ -95,4 +95,2 @@ function getExtensionDownloadStream(extension: IExtensionDefinition) {
 		input = ext.fromGithub(extension, { asset, latest: isInsiders() });
-	} else if (productjson.extensionsGallery?.serviceUrl) {
-		input = ext.fromMarketplace(productjson.extensionsGallery.serviceUrl, extension);
 	} else {
