From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Baptiste Augrain <daiyam@zokugun.org>
Date: Sat, 1 Jan 2000 00:00:00 +0000
Subject: disable the typescript mangler in the build compile task


diff --git a/build/lib/compilation.ts b/build/lib/compilation.ts
index 32e3e25e6bc4302e8328bc4c24553ab8a03be350..6e9396d2f61dd21c011dbbbed7d6aa2e015e6ff7 100644
--- a/build/lib/compilation.ts
+++ b/build/lib/compilation.ts
@@ -137,31 +137,7 @@ export function compileTask(src: string, out: string, build: boolean, options: {
 			generator.execute();
 		}
 
-		// mangle: TypeScript to TypeScript
-		let mangleStream = es.through();
-		if (build && !options.disableMangle) {
-			let ts2tsMangler: Mangler | undefined = new Mangler(compile.projectPath, (...data) => fancyLog(ansiColors.blue('[mangler]'), ...data), { mangleExports: true, manglePrivateFields: true });
-			const newContentsByFileName = ts2tsMangler.computeNewFileContents(new Set(['saveState']));
-			mangleStream = es.through(async function write(data: File & { sourceMap?: RawSourceMap }) {
-				type TypeScriptExt = typeof ts & { normalizePath(path: string): string };
-				const tsNormalPath = (ts as TypeScriptExt).normalizePath(data.path);
-				const newContents = (await newContentsByFileName).get(tsNormalPath);
-				if (newContents !== undefined) {
-					data.contents = Buffer.from(newContents.out);
-					data.sourceMap = newContents.sourceMap && JSON.parse(newContents.sourceMap);
-				}
-				this.push(data);
-			}, async function end() {
-				// free resources
-				(await newContentsByFileName).clear();
-
-				this.push(null);
-				ts2tsMangler = undefined;
-			});
-		}
-
 		const emit = util.streamToPromise(srcPipe
-			.pipe(mangleStream)
 			.pipe(generator.stream)
 			.pipe(compile())
 			.pipe(gulp.dest(out)));
