diff --git a/build/gulpfile.reh.ts b/build/gulpfile.reh.ts
index c643cf99..5688ee60 100644
--- a/build/gulpfile.reh.ts
+++ b/build/gulpfile.reh.ts
@@ -562,3 +562,3 @@ function prepareCopilotRipgrepShimTaskREH(platform: string, arch: string, destin
 
-			const serverTask = task.define(`vscode-${type}${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+			const prepackTask = task.define(`vscode-${type}${dashed(minified)}-prepack`, task.series(
 				copyCodiconsTask,
@@ -569,2 +569,6 @@ function prepareCopilotRipgrepShimTaskREH(platform: string, arch: string, destin
 				compileExtensionMediaBuildTask,
+			));
+			task.task(prepackTask);
+
+			const packingTask = task.define(`vscode-${type}${dashed(platform)}${dashed(arch)}${dashed(minified)}-packing`, task.series(
 				writeISODate('out-build'),
@@ -573,2 +577,8 @@ function prepareCopilotRipgrepShimTaskREH(platform: string, arch: string, destin
 			));
+			task.task(packingTask);
+
+			const serverTask = task.define(`vscode-${type}${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+				prepackTask,
+				packingTask,
+			));
 			task.task(serverTask);
diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
index 95ae1790..28f786e6 100644
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -574,4 +574,4 @@ const BUILD_TARGETS = [
 	{ platform: 'win32', arch: 'arm64' },
-	{ platform: 'darwin', arch: 'x64', opts: { stats: true } },
-	{ platform: 'darwin', arch: 'arm64', opts: { stats: true } },
+	{ platform: 'darwin', arch: 'x64' },
+	{ platform: 'darwin', arch: 'arm64' },
 	{ platform: 'linux', arch: 'x64' },
@@ -584,3 +584,2 @@ BUILD_TARGETS.forEach(buildTarget => {
 	const arch = buildTarget.arch;
-	const opts = buildTarget.opts;
 
@@ -593,3 +592,3 @@ BUILD_TARGETS.forEach(buildTarget => {
 			util.rimraf(path.join(buildRoot, destinationFolderName)),
-			packageTask(platform, arch, sourceFolderName, destinationFolderName, opts),
+			packageTask(platform, arch, sourceFolderName, destinationFolderName),
 			prepareCopilotRipgrepShimTask(platform, arch, destinationFolderName)
@@ -617,3 +616,4 @@ BUILD_TARGETS.forEach(buildTarget => {
 		);
-		const vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+
+		const prepackTask = task.define(`vscode${dashed(minified)}-prepack`, task.series(
 			copyCodiconsTask,
@@ -621,4 +621,7 @@ BUILD_TARGETS.forEach(buildTarget => {
 			compileNonNativeExtensionsBuildTask,
-			compileCopilotExtensionBuildTask,
 			compileExtensionMediaBuildTask,
+		));
+		task.task(prepackTask);
+
+		const packingTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}-packing`, task.series(
 			writeISODate('out-build'),
@@ -627,2 +630,8 @@ BUILD_TARGETS.forEach(buildTarget => {
 		));
+		task.task(packingTask);
+
+		const vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+			prepackTask,
+			packingTask,
+		));
 		task.task(vscodeTask);
diff --git a/build/lib/esbuild.ts b/build/lib/esbuild.ts
index ab0a7021..27ae2de0 100644
--- a/build/lib/esbuild.ts
+++ b/build/lib/esbuild.ts
@@ -56,3 +56,2 @@ export function runEsbuildBundle(outDir: string, minify: boolean, nls: boolean,
 			args.push('--minify');
-			args.push('--mangle-privates');
 		}
