diff --git a/src/vs/platform/extensionManagement/node/extensionDownloader.ts b/src/vs/platform/extensionManagement/node/extensionDownloader.ts
index 0ddae28..2e3c9b2 100644
--- a/src/vs/platform/extensionManagement/node/extensionDownloader.ts
+++ b/src/vs/platform/extensionManagement/node/extensionDownloader.ts
@@ -11,3 +11,2 @@ import { joinPath } from 'vs/base/common/resources';
 import * as semver from 'vs/base/common/semver/semver';
-import { isBoolean } from 'vs/base/common/types';
 import { URI } from 'vs/base/common/uri';
@@ -51,2 +50,3 @@ export class ExtensionsDownloader extends Disposable {
 		@IExtensionGalleryService private readonly extensionGalleryService: IExtensionGalleryService,
+		// @ts-expect-error no-unused-variable
 		@IConfigurationService private readonly configurationService: IConfigurationService,
@@ -116,4 +116,3 @@ export class ExtensionsDownloader extends Disposable {
 
-		const value = this.configurationService.getValue('extensions.verifySignature');
-		return isBoolean(value) ? value : true;
+		return false;
 	}
