diff --git a/build/lib/builtInExtensions.js b/build/lib/builtInExtensions.js
index 400ca68..f37060a 100644
--- a/build/lib/builtInExtensions.js
+++ b/build/lib/builtInExtensions.js
@@ -83,5 +83,3 @@ function isUpToDate(extension) {
 function getExtensionDownloadStream(extension) {
-    const galleryServiceUrl = productjson.extensionsGallery?.serviceUrl;
-    return (galleryServiceUrl ? ext.fromMarketplace(galleryServiceUrl, extension) : ext.fromGithub(extension))
-        .pipe((0, gulp_rename_1.default)(p => p.dirname = `${extension.name}/${p.dirname}`));
+	return ext.fromGithub(extension).pipe((0, gulp_rename_1.default)(p => p.dirname = `${extension.name}/${p.dirname}`));
 }
diff --git a/build/lib/builtInExtensions.ts b/build/lib/builtInExtensions.ts
index 9b1ec73..1cb3301 100644
--- a/build/lib/builtInExtensions.ts
+++ b/build/lib/builtInExtensions.ts
@@ -70,5 +70,3 @@ function isUpToDate(extension: IExtensionDefinition): boolean {
 function getExtensionDownloadStream(extension: IExtensionDefinition) {
-	const galleryServiceUrl = productjson.extensionsGallery?.serviceUrl;
-	return (galleryServiceUrl ? ext.fromMarketplace(galleryServiceUrl, extension) : ext.fromGithub(extension))
-		.pipe(rename(p => p.dirname = `${extension.name}/${p.dirname}`));
+	return ext.fromGithub(extension).pipe(rename(p => p.dirname = `${extension.name}/${p.dirname}`));
 }
