diff --git a/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.ts b/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.ts
index 9564618..35cde81 100644
--- a/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.ts
+++ b/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.ts
@@ -116,2 +116,3 @@ type GettingStartedActionEvent = {
 type RecentEntry = (IRecentFolder | IRecentWorkspace) & { id: string };
+type AnnoucementEntry = { id: string, title: string, url: string };
 
@@ -148,2 +149,4 @@ export class GettingStartedPage extends EditorPane {
 	private gettingStartedList?: GettingStartedIndexList<IResolvedWalkthrough>;
+	private annoucementList?: GettingStartedIndexList<AnnoucementEntry>;
+	private annoucementData?: AnnoucementEntry[];
 
@@ -767,2 +770,3 @@ export class GettingStartedPage extends EditorPane {
 		const gettingStartedList = this.buildGettingStartedWalkthroughsList();
+		const annoucementList = await this.buildAnnoucementList();
 
@@ -777,3 +781,3 @@ export class GettingStartedPage extends EditorPane {
 				this.container.classList.remove('noWalkthroughs');
-				reset(leftColumn, startList.getDomElement(), recentList.getDomElement());
+				reset(leftColumn, startList.getDomElement(), recentList.getDomElement(), annoucementList.getDomElement());
 				reset(rightColumn, gettingStartedList.getDomElement());
@@ -783,3 +787,3 @@ export class GettingStartedPage extends EditorPane {
 				this.container.classList.add('noWalkthroughs');
-				reset(leftColumn, startList.getDomElement());
+				reset(leftColumn, startList.getDomElement(), annoucementList.getDomElement());
 				reset(rightColumn, recentList.getDomElement());
@@ -930,2 +934,41 @@ export class GettingStartedPage extends EditorPane {
 
+	private async buildAnnoucementList(): Promise<GettingStartedIndexList<AnnoucementEntry>> {
+		const renderRecent = (annoucement: AnnoucementEntry) => {
+			const { title, url } = annoucement;
+			const li = $('li');
+
+			const anchor: HTMLLinkElement = $('a');
+			anchor.href = url;
+			anchor.innerText = title;
+			anchor.target = '_blank';
+			li.appendChild(anchor);
+
+			return li;
+		};
+
+		if (this.annoucementList) { this.annoucementList.dispose(); }
+
+		const annoucementList = this.annoucementList = new GettingStartedIndexList({
+			title: localize('annoucements', "VSCodium Annoucements"),
+			klass: 'annoucements',
+			limit: 5,
+			empty: $('.empty-recent', {}, localize('noAnnoucements', "There is no annoucements.")),
+			renderElement: renderRecent,
+			contextService: this.contextService
+		});
+
+		if (!this.annoucementData) {
+			const branch = this.productService.quality === 'insider' ? 'insider' : 'master';
+			const res = await fetch(`https://raw.githubusercontent.com/VSCodium/vscodium/${branch}/annoucements.json`);
+
+			if (res.ok) {
+				this.annoucementData = await res.json() as AnnoucementEntry[];
+			}
+		}
+
+		annoucementList.setEntries(this.annoucementData);
+
+		return annoucementList;
+	}
+
 	private buildStartList(): GettingStartedIndexList<IWelcomePageStartEntry> {
