diff --git a/build/darwin/sign.js b/build/darwin/sign.js
index f5913b7..c861e3e 100644
--- a/build/darwin/sign.js
+++ b/build/darwin/sign.js
@@ -83,3 +83,3 @@ async function main(buildDir) {
             '-string',
-            'An application in Visual Studio Code wants to use AppleScript.',
+            'An application in VSCodium wants to use AppleScript.',
             `${infoPlistPath}`
@@ -90,3 +90,3 @@ async function main(buildDir) {
             '-string',
-            'An application in Visual Studio Code wants to use the Microphone.',
+            'An application in VSCodium wants to use the Microphone.',
             `${infoPlistPath}`
@@ -97,3 +97,3 @@ async function main(buildDir) {
             '-string',
-            'An application in Visual Studio Code wants to use the Camera.',
+            'An application in VSCodium wants to use the Camera.',
             `${infoPlistPath}`
