diff --git a/build/lib/compilation.js b/build/lib/compilation.js
index d101f6e..cf50b50 100644
--- a/build/lib/compilation.js
+++ b/build/lib/compilation.js
@@ -109,3 +109,3 @@ function compileTask(src, out, build) {
             mangleStream = es.through(function write(data) {
-                const newContents = newContentsByFileName.get(data.path);
+                const newContents = newContentsByFileName.get(data.path.replace(/\\/g, '/'));
                 if (newContents !== undefined) {
diff --git a/build/lib/compilation.ts b/build/lib/compilation.ts
index 54fa32d..4c8fd1c 100644
--- a/build/lib/compilation.ts
+++ b/build/lib/compilation.ts
@@ -129,3 +129,3 @@ export function compileTask(src: string, out: string, build: boolean): () => Nod
 			mangleStream = es.through(function write(data: File & { sourceMap?: RawSourceMap }) {
-				const newContents = newContentsByFileName.get(data.path);
+				const newContents = newContentsByFileName.get(data.path.replace(/\\/g, '/'));
 				if (newContents !== undefined) {
