diff --git a/extensions/terminal-suggest/src/completions/codium-insiders.ts b/extensions/terminal-suggest/src/completions/codium-insiders.ts
new file mode 100644
index 0000000..a5769b1
--- /dev/null
+++ b/extensions/terminal-suggest/src/completions/codium-insiders.ts
@@ -0,0 +1,9 @@
+import code from './code';
+
+const codiumInsidersCompletionSpec: Fig.Spec = {
+	...code,
+	name: 'codium-insiders',
+	description: 'VSCodium Insiders',
+};
+
+export default codiumInsidersCompletionSpec;
diff --git a/extensions/terminal-suggest/src/completions/codium.ts b/extensions/terminal-suggest/src/completions/codium.ts
new file mode 100644
index 0000000..b1fa812
--- /dev/null
+++ b/extensions/terminal-suggest/src/completions/codium.ts
@@ -0,0 +1,9 @@
+import code from './code';
+
+const codiumCompletionSpec: Fig.Spec = {
+	...code,
+	name: 'codium',
+	description: 'VSCodium',
+};
+
+export default codiumCompletionSpec;
diff --git a/extensions/terminal-suggest/src/terminalSuggestMain.ts b/extensions/terminal-suggest/src/terminalSuggestMain.ts
index 3f181bc..34054ed 100644
--- a/extensions/terminal-suggest/src/terminalSuggestMain.ts
+++ b/extensions/terminal-suggest/src/terminalSuggestMain.ts
@@ -12,2 +12,4 @@ import codeCompletionSpec from './completions/code';
 import codeInsidersCompletionSpec from './completions/code-insiders';
+import codiumCompletionSpec from './completions/codium';
+import codiumInsidersCompletionSpec from './completions/codium-insiders';
 import setLocationSpec from './completions/set-location';
@@ -52,2 +54,4 @@ export const availableSpecs: Fig.Spec[] = [
 	codeCompletionSpec,
+	codiumInsidersCompletionSpec,
+	codiumCompletionSpec,
 	setLocationSpec,
