--- vscode/src/vs/platform/update/electron-main/updateService.win32.ts	2019-12-08 15:09:44.000000000 -0800
+++ src/src/vs/platform/update/electron-main/updateService.win32.ts	2019-12-08 15:12:56.000000000 -0800
@@ -55,7 +55,7 @@
 
 	@memoize
 	get cachePath(): Promise<string> {
-		const result = path.join(tmpdir(), `vscode-update-${product.target}-${process.arch}`);
+		const result = path.join(tmpdir(), `vscodium-update-${product.target}-${process.arch}`);
 		return pfs.mkdirp(result, undefined).then(() => result);
 	}
 
