diff --git a/build/azure-pipelines/linux/setup-env.sh b/build/azure-pipelines/linux/setup-env.sh
index 2dbbe22a..ddf6488b 100755
--- a/build/azure-pipelines/linux/setup-env.sh
+++ b/build/azure-pipelines/linux/setup-env.sh
@@ -25,16 +25,2 @@ fi
 
-mkdir -p "$HOME/.gyp"
-cat > "$HOME/.gyp/include.gypi" << 'EOF'
-{
-  "target_defaults": {
-    "conditions": [
-      ["OS=='linux'", {
-        'cflags_cc!': [ '-std=gnu++20' ],
-        'cflags_cc': [ '-std=gnu++2a' ],
-      }]
-    ]
-  }
-}
-EOF
-
 if [ "$npm_config_arch" == "x64" ]; then
