diff --git a/build/gulpfile.reh.js b/build/gulpfile.reh.js
index e314794..8248247 100644
--- a/build/gulpfile.reh.js
+++ b/build/gulpfile.reh.js
@@ -437,16 +437,2 @@ function packageTask(type, platform, arch, sourceFolderName, destinationFolderNa
 
-		if (platform === 'linux' && process.env['VSCODE_NODE_GLIBC'] === '-glibc-2.17') {
-			result = es.merge(result,
-				gulp.src(`resources/server/bin/helpers/check-requirements-linux-legacy.sh`, { base: '.' })
-					.pipe(rename(`bin/helpers/check-requirements.sh`))
-					.pipe(util.setExecutableBit())
-			);
-		} else if (platform === 'linux' || platform === 'alpine') {
-			result = es.merge(result,
-				gulp.src(`resources/server/bin/helpers/check-requirements-linux.sh`, { base: '.' })
-					.pipe(rename(`bin/helpers/check-requirements.sh`))
-					.pipe(util.setExecutableBit())
-			);
-		}
-
 		result = inlineMeta(result, {
diff --git a/remote/package.json b/remote/package.json
index 8f29232..1beaa12 100644
--- a/remote/package.json
+++ b/remote/package.json
@@ -33,3 +33,3 @@
     "native-watchdog": "^1.4.1",
-    "node-pty": "1.1.0-beta21",
+    "node-pty": "1.1.0-beta4",
     "tas-client-umd": "0.2.0",
