diff --git a/.yarnrc b/.yarnrc
index 6d69893..5568071 100644
--- a/.yarnrc
+++ b/.yarnrc
@@ -3,3 +3 @@ target "27.2.3"
 ms_build_id "26908389"
-runtime "electron"
-build_from_source "true"
diff --git a/remote/.yarnrc b/remote/.yarnrc
deleted file mode 100644
index cac528f..0000000
--- a/remote/.yarnrc
+++ /dev/null
@@ -1,5 +0,0 @@
-disturl "https://nodejs.org/dist"
-target "16.20.2"
-ms_build_id "255375"
-runtime "node"
-build_from_source "true"
