diff --git a/.yarnrc b/.yarnrc
index 6d69893..849e761 100644
--- a/.yarnrc
+++ b/.yarnrc
@@ -3,3 +3,3 @@ target "27.2.3"
 ms_build_id "26908389"
-runtime "electron"
-build_from_source "true"
+#runtime "electron"
+#build_from_source "true"
diff --git a/remote/.yarnrc b/remote/.yarnrc
index cac528f..0f997ad 100644
--- a/remote/.yarnrc
+++ b/remote/.yarnrc
@@ -3,3 +3,3 @@ target "18.17.1"
 ms_build_id "255375"
-runtime "node"
-build_from_source "true"
+#runtime "node"
+#build_from_source "true"
