diff --git a/src/vs/platform/environment/common/environment.ts b/src/vs/platform/environment/common/environment.ts
index 4fdf7227..fa05f304 100644
--- a/src/vs/platform/environment/common/environment.ts
+++ b/src/vs/platform/environment/common/environment.ts
@@ -141,2 +141,3 @@ export interface INativeEnvironmentService extends IEnvironmentService {
 	userDataPath: string;
+	userStatePath: string;
 
diff --git a/src/vs/platform/environment/common/environmentService.ts b/src/vs/platform/environment/common/environmentService.ts
index 004d0614..7f0af774 100644
--- a/src/vs/platform/environment/common/environmentService.ts
+++ b/src/vs/platform/environment/common/environmentService.ts
@@ -28,2 +28,4 @@ export interface INativeEnvironmentPaths {
 
+	userStateDir: string;
+
 	/**
@@ -54,2 +56,5 @@ export abstract class AbstractNativeEnvironmentService implements INativeEnviron
 
+	@memoize
+	get userStatePath(): string { return this.paths.userStateDir; }
+
 	@memoize
@@ -75,3 +80,3 @@ export abstract class AbstractNativeEnvironmentService implements INativeEnviron
 			const key = toLocalISOString(new Date()).replace(/-|:|\.\d+Z$/g, '');
-			this.args.logsPath = join(this.userDataPath, 'logs', key);
+			this.args.logsPath = join(this.userStatePath, key);
 		}
diff --git a/src/vs/platform/environment/node/environmentService.ts b/src/vs/platform/environment/node/environmentService.ts
index 8652144b..62fc4ea6 100644
--- a/src/vs/platform/environment/node/environmentService.ts
+++ b/src/vs/platform/environment/node/environmentService.ts
@@ -11,2 +11,3 @@ import { getUserDataPath } from './userDataPath.js';
 import { IProductService } from '../../product/common/productService.js';
+import { getUserStatePath } from './userStatePath.js';
 
@@ -20,2 +21,3 @@ export class NativeEnvironmentService extends AbstractNativeEnvironmentService {
 			userDataDir: getUserDataPath(args, productService.nameShort),
+			userStateDir: getUserStatePath(args, productService.nameShort),
 		}, productService);
diff --git a/src/vs/platform/environment/node/userStatePath.ts b/src/vs/platform/environment/node/userStatePath.ts
new file mode 100644
index 00000000..53f9e2f1
--- /dev/null
+++ b/src/vs/platform/environment/node/userStatePath.ts
@@ -0,0 +1,42 @@
+import { homedir } from 'os';
+import { NativeParsedArgs } from '../common/argv.js';
+
+import { resolve, isAbsolute, join } from 'path';
+
+const cwd = process.env['VSCODE_CWD'] || process.cwd();
+
+export function getUserStatePath(cliArgs: NativeParsedArgs, productName: string): string {
+	const userStatePath = doGetUserStatePath(cliArgs, productName);
+	const pathsToResolve = [userStatePath];
+
+	if (!isAbsolute(userStatePath)) {
+		pathsToResolve.unshift(cwd);
+	}
+
+	return resolve(...pathsToResolve);
+}
+
+function doGetUserStatePath(cliArgs: NativeParsedArgs, productName: string): string {
+
+	// 0. Running out of sources has a fixed productName
+	if (process.env['VSCODE_DEV']) {
+		productName = 'code-oss-dev';
+	}
+
+	// 1. Support portable mode
+	const portablePath = process.env['VSCODE_PORTABLE'];
+	if (portablePath) {
+		return join(portablePath, 'user-state');
+	}
+
+	// 2. Support global VSCODE_APPSTATE environment variable
+	let appStatePath = process.env['VSCODE_APPSTATE'];
+	if (appStatePath) {
+		return join(appStatePath, productName);
+	}
+
+	// 4. Otherwise
+	appStatePath = process.env['XDG_STATE_HOME'] || join(homedir(), '.local', 'state');
+
+	return join(appStatePath, productName);
+}
diff --git a/src/vs/platform/userDataProfile/test/common/userDataProfileService.test.ts b/src/vs/platform/userDataProfile/test/common/userDataProfileService.test.ts
index 37cd505a..9ae46919 100644
--- a/src/vs/platform/userDataProfile/test/common/userDataProfileService.test.ts
+++ b/src/vs/platform/userDataProfile/test/common/userDataProfileService.test.ts
@@ -24,4 +24,6 @@ class TestEnvironmentService extends AbstractNativeEnvironmentService {
 		const userDataDir = _appSettingsHome.fsPath.replace(/\/User$/, '');
+		const userStateDir = _appSettingsHome.fsPath.replace(/\/State/, '');
 		const paths: INativeEnvironmentPaths = {
 			userDataDir,
+			userStateDir,
 			homeDir: userDataDir,
diff --git a/src/vs/platform/userDataProfile/test/electron-main/userDataProfileMainService.test.ts b/src/vs/platform/userDataProfile/test/electron-main/userDataProfileMainService.test.ts
index c8ec2ae6..e6ad98a3 100644
--- a/src/vs/platform/userDataProfile/test/electron-main/userDataProfileMainService.test.ts
+++ b/src/vs/platform/userDataProfile/test/electron-main/userDataProfileMainService.test.ts
@@ -25,4 +25,6 @@ class TestEnvironmentService extends AbstractNativeEnvironmentService {
 		const userDataDir = _appSettingsHome.fsPath.replace(/\/User$/, '');
+		const userStateDir = _appSettingsHome.fsPath.replace(/\/State/, '');
 		const paths: INativeEnvironmentPaths = {
 			userDataDir,
+			userStateDir,
 			homeDir: userDataDir,
diff --git a/src/vs/platform/window/common/window.ts b/src/vs/platform/window/common/window.ts
index 291648bc..29c28d21 100644
--- a/src/vs/platform/window/common/window.ts
+++ b/src/vs/platform/window/common/window.ts
@@ -444,2 +444,3 @@ export interface INativeWindowConfiguration extends IWindowConfiguration, Native
 	userDataDir: string;
+	userStateDir: string;
 
diff --git a/src/vs/platform/windows/electron-main/windowsMainService.ts b/src/vs/platform/windows/electron-main/windowsMainService.ts
index 9feeb83e..53975c9f 100644
--- a/src/vs/platform/windows/electron-main/windowsMainService.ts
+++ b/src/vs/platform/windows/electron-main/windowsMainService.ts
@@ -1562,2 +1562,3 @@ export class WindowsMainService extends Disposable implements IWindowsMainServic
 			userDataDir: this.environmentMainService.userDataPath,
+			userStateDir: this.environmentMainService.userStatePath,
 
diff --git a/src/vs/workbench/services/environment/electron-browser/environmentService.ts b/src/vs/workbench/services/environment/electron-browser/environmentService.ts
index 1abd21a9..eafaaa1f 100644
--- a/src/vs/workbench/services/environment/electron-browser/environmentService.ts
+++ b/src/vs/workbench/services/environment/electron-browser/environmentService.ts
@@ -166,2 +166,3 @@ export class NativeWorkbenchEnvironmentService extends AbstractNativeEnvironment
 				userDataDir: configuration.userDataDir,
+				userStateDir: configuration.userStateDir,
 			},
diff --git a/src/vs/workbench/services/workingCopy/test/electron-browser/workingCopyBackupService.test.ts b/src/vs/workbench/services/workingCopy/test/electron-browser/workingCopyBackupService.test.ts
index 7420d5e5..767502b0 100644
--- a/src/vs/workbench/services/workingCopy/test/electron-browser/workingCopyBackupService.test.ts
+++ b/src/vs/workbench/services/workingCopy/test/electron-browser/workingCopyBackupService.test.ts
@@ -76,2 +76,3 @@ const TestNativeWindowConfiguration: INativeWindowConfiguration = {
 	userDataDir: joinPath(homeDir, product.nameShort).fsPath,
+	userStateDir: joinPath(homeDir, product.nameShort).fsPath,
 	profiles: { profile: NULL_PROFILE, all: [NULL_PROFILE], home: homeDir },
