diff --git a/src/bootstrap-esm.ts b/src/bootstrap-esm.ts
index 49098a31..ae041b7d 100644
--- a/src/bootstrap-esm.ts
+++ b/src/bootstrap-esm.ts
@@ -6,3 +6,3 @@
 import * as fs from 'node:fs';
-import { register } from 'node:module';
+import * as Module from 'node:module';
 import { dirname, join } from 'node:path';
@@ -40,3 +40,3 @@ globalThis._VSCODE_FILE_ROOT = import.meta.dirname;
 function enableASARSupport(): void {
-	if (!process.env['ELECTRON_RUN_AS_NODE'] && !process.versions['electron']) {
+	if (Module.register as unknown && (process.env['ELECTRON_RUN_AS_NODE'] || process.versions['electron'])) {
 		return; // only on Electron / Electron-as-node
@@ -235,3 +235,3 @@ function enableASARSupport(): void {
 
-	register(`data:text/javascript;base64,${Buffer.from(jsCode).toString('base64')}`, import.meta.url, {
+	Module.register(`data:text/javascript;base64,${Buffer.from(jsCode).toString('base64')}`, import.meta.url, {
 		data: process.env['VSCODE_DEV'] ? {} : {
