diff --git a/build/linux/debian/calculate-deps.ts b/build/linux/debian/calculate-deps.ts
index 4597dab7..bce93a37 100644
--- a/build/linux/debian/calculate-deps.ts
+++ b/build/linux/debian/calculate-deps.ts
@@ -113,2 +113,4 @@ function calculatePackageDeps(binaryPath: string, arch: DebianArchString, chromi
 		return !dependency.startsWith('libgcc-s1');
+	}).map(dependency => {
+		return dependency.startsWith('libasound2 (>= 1.0.17)') ? 'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2' : dependency;
 	}).sort();
diff --git a/build/linux/debian/dep-lists.ts b/build/linux/debian/dep-lists.ts
index 37e04515..06b8f0be 100644
--- a/build/linux/debian/dep-lists.ts
+++ b/build/linux/debian/dep-lists.ts
@@ -27,3 +27,3 @@ export const referenceGeneratedDepsByArch = {
 		'ca-certificates',
-		'libasound2 (>= 1.0.17)',
+		'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2',
 		'libatk-bridge2.0-0 (>= 2.5.3)',
@@ -66,3 +66,3 @@ export const referenceGeneratedDepsByArch = {
 		'ca-certificates',
-		'libasound2 (>= 1.0.17)',
+		'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2',
 		'libatk-bridge2.0-0 (>= 2.5.3)',
@@ -109,3 +109,3 @@ export const referenceGeneratedDepsByArch = {
 		'ca-certificates',
-		'libasound2 (>= 1.0.17)',
+		'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2',
 		'libatk-bridge2.0-0 (>= 2.5.3)',
@@ -149,3 +149,3 @@ export const referenceGeneratedDepsByArch = {
 		'ca-certificates',
-		'libasound2 (>= 1.0.17)',
+		'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2',
 		'libatk-bridge2.0-0 (>= 2.5.3)',
@@ -189,3 +189,3 @@ export const referenceGeneratedDepsByArch = {
 		'libatomic1',
-		'libasound2 (>= 1.0.17)',
+		'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2',
 		'libatk-bridge2.0-0 (>= 2.5.3)',
@@ -229,3 +229,3 @@ export const referenceGeneratedDepsByArch = {
 		'libatomic1',
-		'libasound2 (>= 1.0.17)',
+		'libasound2t64 (>= 1.2.12) | libasound2 (>= 1.0.17) | liboss4-salsa-asound2',
 		'libatk-bridge2.0-0 (>= 2.5.3)',
