diff --git a/cli/src/tunnels/code_server.rs b/cli/src/tunnels/code_server.rs
index bbabadc..08af0df 100644
--- a/cli/src/tunnels/code_server.rs
+++ b/cli/src/tunnels/code_server.rs
@@ -4,2 +4,4 @@
  *--------------------------------------------------------------------------------------------*/
+#![allow(unused_imports)]
+
 use super::paths::{InstalledServer, ServerPaths};
diff --git a/cli/src/tunnels/dev_tunnels.rs b/cli/src/tunnels/dev_tunnels.rs
index 0168ee6..5d0d8d4 100644
--- a/cli/src/tunnels/dev_tunnels.rs
+++ b/cli/src/tunnels/dev_tunnels.rs
@@ -4,2 +4,4 @@
  *--------------------------------------------------------------------------------------------*/
+#![allow(unused_imports)]
+
 use super::protocol::{self, PortPrivacy, PortProtocol};
diff --git a/cli/src/update_service.rs b/cli/src/update_service.rs
index 55f1dad..f22a83c 100644
--- a/cli/src/update_service.rs
+++ b/cli/src/update_service.rs
@@ -4,2 +4,3 @@
  *--------------------------------------------------------------------------------------------*/
+#![allow(unused_imports)]
 
