diff --git a/src/vs/base/common/policy.ts b/src/vs/base/common/policy.ts
index 8141b0f..33599d9 100644
--- a/src/vs/base/common/policy.ts
+++ b/src/vs/base/common/policy.ts
@@ -11,3 +11,3 @@ import { IDefaultAccount } from './defaultAccount.js';
  */
-export const LINUX_SYSTEM_POLICY_FILE_PATH = '/etc/vscode/policy.json';
+export const LINUX_SYSTEM_POLICY_FILE_PATH = '/etc/!!GLOBAL_DIRNAME!!/policy.json';
 
