diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index fa8da7d..abd8957 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -55,5 +55,7 @@ function npmInstall(dir, opts) {
 		opts.cwd = root;
-		if (process.env['npm_config_arch'] === 'arm64') {
-			run('sudo', ['docker', 'run', '--rm', '--privileged', 'multiarch/qemu-user-static', '--reset', '-p', 'yes'], opts);
-		}
+		if ((process.env['npm_config_arch'] === 'arm64' || process.env['npm_config_arch'] === 'arm' || process.env['npm_config_arch'] === 'ppc64' || process.env['npm_config_arch'] === 'riscv64' || process.env['npm_config_arch'] === 's390x') && process.env['DISABLE_QEMU'] !== 'true') {
+			run('sudo', ['docker', 'run', '--rm', '--privileged', 'tonistiigi/binfmt:latest', '--install', 'all'], opts);
+		} else if (process.env['npm_config_arch'] === 'loong64') {
+			run('sudo', ['docker', 'run', '--rm', '--privileged', 'loongcr.lcpu.dev/multiarch/archlinux', '--reset', '-p', 'yes'], opts);
+ 		}
 		run('sudo', [
