diff --git a/build/gulpfile.reh.js b/build/gulpfile.reh.js
index df18cf0..7f485a0 100644
--- a/build/gulpfile.reh.js
+++ b/build/gulpfile.reh.js
@@ -374,10 +374,2 @@ function packageTask(type, platform, arch, sourceFolderName, destinationFolderNa
 
-		if (platform === 'linux' || platform === 'alpine') {
-			result = es.merge(result,
-				gulp.src(`resources/server/bin/helpers/check-requirements-linux.sh`, { base: '.' })
-					.pipe(rename(`bin/helpers/check-requirements.sh`))
-					.pipe(util.setExecutableBit())
-			);
-		}
-
 		return result.pipe(vfs.dest(destination));
diff --git a/remote/.yarnrc b/remote/.yarnrc
index adbe2d2..dfc3bd7 100644
--- a/remote/.yarnrc
+++ b/remote/.yarnrc
@@ -1,3 +1,3 @@
 disturl "https://nodejs.org/dist"
-target "18.17.1"
+target "16.20.2"
 ms_build_id "252256"
diff --git a/remote/package.json b/remote/package.json
index b1fbc7b..87716c0 100644
--- a/remote/package.json
+++ b/remote/package.json
@@ -31,3 +31,3 @@
     "native-watchdog": "^1.4.1",
-    "node-pty": "1.1.0-beta5",
+    "node-pty": "1.1.0-beta4",
     "tas-client-umd": "0.1.8",
diff --git a/remote/yarn.lock b/remote/yarn.lock
index bb20531..8f7c051 100644
--- a/remote/yarn.lock
+++ b/remote/yarn.lock
@@ -433,6 +433,6 @@ node-gyp-build@^4.3.0:
 
-node-pty@1.1.0-beta5:
-  version "1.1.0-beta5"
-  resolved "https://registry.yarnpkg.com/node-pty/-/node-pty-1.1.0-beta5.tgz#364386b7058a93070234064f13164ec1ef914993"
-  integrity sha512-j3QdgFHnLY0JWxztrvM3g67RaQLOGvytv+C6mFu0PqD+JILlzqfwuoyqRqVxdZZjoOTUXPfSRj1qPVCaCH+eOw==
+node-pty@1.1.0-beta4:
+  version "1.1.0-beta4"
+  resolved "https://registry.yarnpkg.com/node-pty/-/node-pty-1.1.0-beta4.tgz#ee74d909c9f422ffc7f675e1092529673f8906ec"
+  integrity sha512-CgffN9AxVtH4g7vDxtanm2qaR7jw3oet9r+ArzziGiFvmds9SdR3gXkZF0fqZWSxhTHZusJWvsuKvRv+5O2K8A==
   dependencies:
diff --git a/resources/server/bin/code-server-linux.sh b/resources/server/bin/code-server-linux.sh
index e3d96bd..3df32df 100644
--- a/resources/server/bin/code-server-linux.sh
+++ b/resources/server/bin/code-server-linux.sh
@@ -11,22 +11,2 @@ ROOT="$(dirname "$(dirname "$(readlink -f "$0")")")"
 
-# Do not remove this check.
-# Provides a way to skip the server requirements check from
-# outside the install flow. A system process can create this
-# file before the server is downloaded and installed.
-skip_check=0
-if [ -f "/tmp/vscode-skip-server-requirements-check" ]; then
-	echo "!!! WARNING: Skipping server pre-requisite check !!!"
-	echo "!!! Server stability is not guaranteed. Proceed at your own risk. !!!"
-	skip_check=1
-fi
-
-# Check platform requirements
-if [ "$(echo "$@" | grep -c -- "--skip-requirements-check")" -eq 0 ] && [ $skip_check -eq 0 ]; then
-	$ROOT/bin/helpers/check-requirements.sh
-	exit_code=$?
-	if [ $exit_code -ne 0 ]; then
-		exit $exit_code
-	fi
-fi
-
 "$ROOT/node" ${INSPECT:-} "$ROOT/out/server-main.js" "$@"
