diff --git a/resources/linux/rpm/code.spec.template b/resources/linux/rpm/code.spec.template
index a73bc02..242bab0 100644
--- a/resources/linux/rpm/code.spec.template
+++ b/resources/linux/rpm/code.spec.template
@@ -2,3 +2,3 @@ Name:     @@NAME@@
 Version:  @@VERSION@@
-Release:  @@RELEASE@@.el8
+Release:  el9
 Summary:  Code editing. Redefined.
@@ -88 +88,3 @@ update-mime-database %{_datadir}/mime &> /dev/null || :
 %{_datadir}/zsh/site-functions/_%{name}
+
+%config(noreplace) /usr/share/%{name}/resources/app/product.json
\ No newline at end of file
