diff --git a/build/checksums/vscode-sysroot.txt b/build/checksums/vscode-sysroot.txt
index 86da04d..0b5f38c 100644
--- a/build/checksums/vscode-sysroot.txt
+++ b/build/checksums/vscode-sysroot.txt
@@ -1,3 +1,6 @@
-a2d51dc505ed544c52757f90bcdab44920132295fc7a67166eff86b6e0e24aa8  aarch64-linux-gnu.tar.gz
-032cf16bf8b965e1351305f10f3dedabf4f9868027ac6d0e8f52321ca0b70d4a  arm-rpi-linux-gnueabihf.tar.gz
-360475a764d0faf4d3743aa866347eff78072639d20660def83e1a03eadf534c  x86_64-linux-gnu.tar.gz
+68a17006021975ff271a1dd615f9db9eda7c25f2cc65e750c87980dc57a06c94  aarch64-linux-gnu-glibc-2.17.tar.gz
+0de422a81683cf9e8cf875dbd1e0c27545ac3c775b2d53015daf3ca2b31d3f15  aarch64-linux-gnu-glibc-2.28.tar.gz
+3ced48cb479f2cdba95aa649710fcb7778685551c745bbd76ac706c3c0ead9fb  arm-rpi-linux-gnueabihf-glibc-2.17.tar.gz
+7aea163f7fad8cc50000c86b5108be880121d35e2f55d016ef8c96bbe54129eb  arm-rpi-linux-gnueabihf-glibc-2.28.tar.gz
+5aae21115f1d284c3cdf32c83db15771b59bc80793f1423032abf5a823c0d658  x86_64-linux-gnu-glibc-2.17.tar.gz
+dbb927408393041664a020661f2641c9785741be3d29b050b9dac58980967784  x86_64-linux-gnu-glibc-2.28.tar.gz
diff --git a/build/linux/debian/install-sysroot.js b/build/linux/debian/install-sysroot.js
index 40ca42e..d637fce 100644
--- a/build/linux/debian/install-sysroot.js
+++ b/build/linux/debian/install-sysroot.js
@@ -69,3 +69,3 @@ async function fetchUrl(options, retries = 10, retryDelay = 1000) {
         const timeout = setTimeout(() => controller.abort(), 30 * 1000);
-        const version = '20231122-245579';
+        const version = '20240129-253798';
         try {
@@ -121,5 +121,6 @@ async function getVSCodeSysroot(arch) {
     let triple;
+    const prefix = process.env['VSCODE_SYSROOT_PREFIX'] ?? '-glibc-2.28';
     switch (arch) {
         case 'amd64':
-            expectedName = `x86_64-linux-gnu.tar.gz`;
+            expectedName = `x86_64-linux-gnu${prefix}.tar.gz`;
             triple = 'x86_64-linux-gnu';
@@ -127,3 +128,3 @@ async function getVSCodeSysroot(arch) {
         case 'arm64':
-            expectedName = `aarch64-linux-gnu.tar.gz`;
+            expectedName = `aarch64-linux-gnu${prefix}.tar.gz`;
             triple = 'aarch64-linux-gnu';
@@ -131,3 +132,3 @@ async function getVSCodeSysroot(arch) {
         case 'armhf':
-            expectedName = `arm-rpi-linux-gnueabihf.tar.gz`;
+            expectedName = `arm-rpi-linux-gnueabihf${prefix}.tar.gz`;
             triple = 'arm-rpi-linux-gnueabihf';
@@ -135,2 +136,3 @@ async function getVSCodeSysroot(arch) {
     }
+    console.log(`Fetching ${expectedName} for ${triple}`);
     const checksumSha256 = getVSCodeSysrootChecksum(expectedName);
diff --git a/build/linux/rpm/dep-lists.js b/build/linux/rpm/dep-lists.js
index e824f19..b9a6e80 100644
--- a/build/linux/rpm/dep-lists.js
+++ b/build/linux/rpm/dep-lists.js
@@ -44,6 +44,3 @@ exports.referenceGeneratedDepsByArch = {
         'libc.so.6(GLIBC_2.17)(64bit)',
-        'libc.so.6(GLIBC_2.18)(64bit)',
         'libc.so.6(GLIBC_2.2.5)(64bit)',
-        'libc.so.6(GLIBC_2.25)(64bit)',
-        'libc.so.6(GLIBC_2.27)(64bit)',
         'libc.so.6(GLIBC_2.28)(64bit)',
@@ -143,5 +140,2 @@ exports.referenceGeneratedDepsByArch = {
         'libc.so.6(GLIBC_2.17)',
-        'libc.so.6(GLIBC_2.18)',
-        'libc.so.6(GLIBC_2.25)',
-        'libc.so.6(GLIBC_2.27)',
         'libc.so.6(GLIBC_2.28)',
@@ -247,5 +241,2 @@ exports.referenceGeneratedDepsByArch = {
         'libc.so.6(GLIBC_2.17)(64bit)',
-        'libc.so.6(GLIBC_2.18)(64bit)',
-        'libc.so.6(GLIBC_2.25)(64bit)',
-        'libc.so.6(GLIBC_2.27)(64bit)',
         'libc.so.6(GLIBC_2.28)(64bit)',
