diff --git a/build/lib/fetch.js b/build/lib/fetch.js
index ba23e78..9b8a5a1 100644
--- a/build/lib/fetch.js
+++ b/build/lib/fetch.js
@@ -13,2 +13,3 @@ const crypto = require("crypto");
 const through2 = require("through2");
+const fetch = require("node-fetch");
 function fetchUrls(urls, options) {
diff --git a/build/linux/debian/install-sysroot.js b/build/linux/debian/install-sysroot.js
index 40ca42e..3276167 100644
--- a/build/linux/debian/install-sysroot.js
+++ b/build/linux/debian/install-sysroot.js
@@ -14,2 +14,3 @@ const crypto_1 = require("crypto");
 const ansiColors = require("ansi-colors");
+const fetch = require("node-fetch");
 // Based on https://source.chromium.org/chromium/chromium/src/+/main:build/linux/sysroot_scripts/install-sysroot.py.
diff --git a/remote/.yarnrc b/remote/.yarnrc
index adbe2d2..dfc3bd7 100644
--- a/remote/.yarnrc
+++ b/remote/.yarnrc
@@ -1,3 +1,3 @@
 disturl "https://nodejs.org/dist"
-target "18.17.1"
+target "16.20.2"
 ms_build_id "252256"
