diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index 88e3c9e..4cad4f1 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -55,4 +55,6 @@ function npmInstall(dir, opts) {
 		opts.cwd = root;
-		if (process.env['npm_config_arch'] === 'arm64') {
+		if (process.env['npm_config_arch'] === 'arm64' || process.env['npm_config_arch'] === 'arm' || process.env['npm_config_arch'] === 'ppc64' || process.env['npm_config_arch'] === 'riscv64' || process.env['npm_config_arch'] === 's390x') {
 			run('sudo', ['docker', 'run', '--rm', '--privileged', 'multiarch/qemu-user-static', '--reset', '-p', 'yes'], opts);
+		} else if (process.env['npm_config_arch'] === 'loong64') {
+			run('sudo', ['docker', 'run', '--rm', '--privileged', 'loongcr.lcpu.dev/multiarch/archlinux', '--reset', '-p', 'yes'], opts);
 		}
