diff --git a/build/azure-pipelines/linux/install.sh b/build/azure-pipelines/linux/install.sh
index 57f5876..2d543e5 100755
--- a/build/azure-pipelines/linux/install.sh
+++ b/build/azure-pipelines/linux/install.sh
@@ -7,10 +7,12 @@ yarn config set registry "$NPM_REGISTRY"
 
-SYSROOT_ARCH=$VSCODE_ARCH
-if [ "$SYSROOT_ARCH" == "x64" ]; then
-  SYSROOT_ARCH="amd64"
+if [ "$npm_config_arch" != "ppc64" ]; then
+	SYSROOT_ARCH=$VSCODE_ARCH
+	if [ "$SYSROOT_ARCH" == "x64" ]; then
+		SYSROOT_ARCH="amd64"
+	fi
+
+	export VSCODE_SYSROOT_DIR=$PWD/.build/sysroots
+	SYSROOT_ARCH="$SYSROOT_ARCH" node -e '(async () => { const { getVSCodeSysroot } = require("./build/linux/debian/install-sysroot.js"); await getVSCodeSysroot(process.env["SYSROOT_ARCH"]); })()'
 fi
 
-export VSCODE_SYSROOT_DIR=$PWD/.build/sysroots
-SYSROOT_ARCH="$SYSROOT_ARCH" node -e '(async () => { const { getVSCodeSysroot } = require("./build/linux/debian/install-sysroot.js"); await getVSCodeSysroot(process.env["SYSROOT_ARCH"]); })()'
-
 if [ "$npm_config_arch" == "x64" ]; then
